core/src/main/resources/error/error-classes.json

```diff
@@ -592,6 +592,17 @@
       "Detected an incompatible DataSourceRegister. Please remove the incompatible library from classpath or upgrade it. Error: <message>"
     ]
   },
+  "INCOMPATIBLE_TEMP_VIEW_SCHEMA_CHANGE" : {
```

Comment:
I propose to merge these two. Presumably the causes are exactly the same.
As noted in the ticket I also propose to merge all causes of a re-create of a view failing into one error class and adding subclasses for the distinct causes.
I think that would be:
* CANNOT UPCAST
* TABLE NOT FOUND/UNRESOLVED
* UNRESOLVED FUNCTION
* DATATYPE MISMATCH

Really it's a nested error...

Reply:
Sounds good, let me merge them into single error class.
Also, just create ticket SPARK-42165 to merge the re-create of a view failing errors.
Let me also create a PR for this ticket soon.